Feature/webserver

sskapci opened the pull request.

sskapci committed:
The first working version

sskapci committed:
WIP
offset management started
Not working version
Lock might be needed for offset management
offset management should be an arrow file

sskapci committed:
REQUIREMENTS.md has changed

sskapci committed:
Create DATA_GOVERNANCE.md

sskapci committed:
Not working version.
This is for offset topic

sskapci committed:
Merge remote-tracking branch 'origin/feature/webserver' into feature/webserver
Merge

sskapci committed:
Still Not working version.
Ownership issues

sskapci committed:
Ownership problem has been solved.
The file issue persist

sskapci committed:
Todos updated
Offset mechanism is still WIP
Concurency is an issue for the time being

sskapci committed:
Something is working about arrow.
Offset management is still a burning issue

sskapci committed:
Refactoring
Offset mechanism is still WIP
Concurency is an issue for the time being

sskapci committed:
Offset mechanism is still WIP
Arrow didn't work well

sskapci committed:
Offset mechanism is still WIP
in memory management

sskapci committed:
Offset mechanism seems working
in memory management

sskapci committed:
Shared Mutex
readers-writer lock

sskapci committed:
the beginning of the web server

sskapci committed:
Documentation has been added

sskapci committed:
Documentation has been added

sskapci committed:
WIP
Seastar REST seems promising
TCP is not there yet

sskapci committed:
WIP
Seastar REST is working.
TCP is not there yet

sskapci committed:
WIP
Seastar REST is working. Initial security R&D started. Basic Auth only
TCP is not there yet

sskapci committed:
WIP
Seastar REST is working. Initial security R&D started. Basic Auth only.
New discovery, partitioner logic is needed.
TCP is not there yet

sskapci committed:
WIP
Seastar REST is working. Initial security R&D started. Basic Auth only.
New discovery, partitioner logic is needed.
TCP is not there yet

billy958 commented on http/message_store.h:
```diff
@@ -0,0 +1,55 @@
+class message_store {
+public:
+    std::unordered_map<int, std::string> messages;
+    int next_id = 0;
```
I know it's a prototype but consider:
1. making it `unsigned int` (I think you don't plan to store the negative values in here because you initialized it at 0 and only increased it later)
2. the int, for now, is large enough, but it might be `unsigned long long` later.

sskapci replied:
Converted to unsigned int for the time being. This temp store should be gone shortly.

sskapci committed:
Not Working version

Partitioner logic is kinda done. Round robin only
Back to offset management.
TCP is not there yet

sskapci committed:
Not Working version

Parameterization has just started.
Partitioner logic is kinda done. Round robin only
Back to offset management.
TCP is not there yet

sskapci committed:
Not Working version

Locks are failing!
Parameterization has just started.
Partitioner logic is kinda done. Round robin only
Back to offset management.
TCP is not there yet

sskapci committed:
Working version

Locks are bypassed for some high level things! needs to be revisited
Partitioner logic is kinda done. Round robin only
Back to offset management.
TCP is not there yet

sskapci committed:
Working version

Locks are bypassed for some high level things! needs to be revisited
Partitioner logic is kinda done. Round robin only
Back to offset management.
TCP is not there yet

sskapci committed:
Working version with virtual TCP

Locks are bypassed for some high level things! needs to be revisited
Partitioner logic is kinda done. Round robin only
Back to offset management.
TCP is virtually working

sskapci committed:
Working version with TCP.
TCP and webservice are sharing same message store now

Locks are bypassed for some high level things! needs to be revisited
Partitioner logic is kinda done. Round robin only

sskapci committed:
Working version with TCP.
TCP and webservice are sharing same message store now

Locks are bypassed for some high level things! needs to be revisited
Partitioner logic is kinda done. Round robin only
